### ISSUE TITLE
Ensure uniform representation of data across config and cluster

### ISSUE THREAD

**Issue by mohammedzee1000** [openshift/odo#1737]
[kind/Enhancement]

### TL;DR

Consider `obj` (which can be component, url, storage etc) which can be in 2 places `config` and `cluster`

Currently, json output of such `obj` type will have `obj` if it is in `cluster`

For normal output details of `obj` are only given if its in `config` with an indication if it is in `cluster` or not
Basically `obj` in `cluster` but not in `config` are ignored

### if json flag is set 
```
	if o.OutputFlag == "json" {
		out, err := json.Marshal(urls)
		if err != nil {
			return err
		}
		fmt.Println(string(out))
	} 
```
Basically urls on cluster are simply printed
### If it is not set
```
		if len(urls.Items) == 0 && len(localUrls) == 0 {
			return fmt.Errorf("no URLs found for component %v in application %v", o.Component(), o.Application)
		} else {
			ppm := false
			log.Infof("Found the following URLs for component %v in application %v:", o.Component(), o.Application)
			tabWriterURL := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
			//create headers
			fmt.Fprintln(tabWriterURL, "NAME", "\t", "URL", "\t", "PORT")

			for _, i := range localUrls {
				var present bool
				for _, u := range urls.Items {
					if i.Name == u.Name {
						fmt.Fprintln(tabWriterURL, u.Name, "\t", url.GetURLString(u.Spec.Protocol, u.Spec.Host), "\t", u.Spec.Port)
						present = true
					}
				}
				if !present {
					ppm = true
					fmt.Fprintln(tabWriterURL, i.Name, "\t", "<not created on cluster>", "\t", i.Port)
				}
			}

			tabWriterURL.Flush()
			if ppm {
				genericclioptions.PrintPushMessage("URL")
			}
```
Basially only urls present in the config and then present on the cluster are printed.

This is not just in URLs but can be seen across the board in all of our objects, such as storage, component etc

The final data that is presented should be unified,

## Why is this needed? 
Someone depending on json data ends up not getting the full information
Update1:
Also, what about stuff in cluster, not in config(removed from config but not pushed to cluster)?
